### PR TITLE
Dev env script

### PR DESCRIPTION
**Pull request by ewjoachim**
Closes no ticket

Adds a script that takes care of everything (venv, db, ...) when contributing to the repo.

### Successful PR Checklist:

- [ ] Tests
  - [x] (not applicable?)
- [x] Documentation
  - [ ] (not applicable?)
- [x] Had a good time contributing?
- [ ] (Maintainers: add PR labels)

### CODE REVIEW

File: dev-env [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+if [ -z "$VIRTUAL_ENV" ]; then
+    if [ ! -d ".venv" ]; then
+        echo "Creating virtual environment at .venv"
+        python3 -m venv .venv
+    fi
+    echo "Activating virtual environment at .venv"
+    source .venv/bin/activate
+fi
+
+echo "Virtualenv is present and active !"
```
Comment (elemoine) [inline dev-env:15]:
```suggestion
echo "Virtualenv is present and active!"
```

File: dev-env [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+if ! ls .venv/lib/python*/site-packages/procrastinate.egg-link 2>&1 1>/dev/null ; then
```
Comment (elemoine) [inline dev-env:17]: Don't you want that both file descriptions 2 and 1 to go to /dev/null? In that case I guess you want to use `1>/dev/null 2>&1`.

File: dev-env [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+if ! pg_isready; then
+    echo "Starting database"
+    export PGDATABASE=procrastinate PGHOST=localhost PGUSER=postgres PGPASSWORD=password
+    docker-compose up -d postgres
+    sleep 3
+    echo "Applying migrations"
+    procrastinate schema --apply
+fi
+
+echo "Database is ready !"
```
Comment (elemoine) [inline dev-env:48]:
```suggestion
echo "Database is ready!"
```

File: dev-env [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+alias docs='tox -e docs'
+
+echo ""
+echo "Welcome to the Procrastinate development environment interactive shell!"
+echo ""
+echo "You'll find the detailed instructions in the contributing documentation:"
+echo "    https://procrastinate.readthedocs.io/en/latest/contributing.html"
+echo ""
+echo "TL;DR: important commands:"
+echo "- pytest : Launch the tests"
```
Comment (elemoine) [inline dev-env:65]:
```suggestion
echo "- pytest: Launch the tests"
```

File: dev-env [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+echo "- tox : Entrypoint for testing multiple python ves as well as docs, linters & formatters"
```
Comment (elemoine) [inline dev-env:66]:
```suggestion
echo "- tox: Entrypoint for testing multiple python ves as well as docs, linters & formatters"
```

Comment (elemoine) [reply dev-env:66]: ves?

File: dev-env [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+echo "- procrastinate: Test procrastinate locally."
+echo ""
+echo "We've gone ahead and setup a few additional commands for you:"
```
Comment (elemoine) [inline dev-env:69]:
```suggestion
echo "We've gone ahead and set up a few additional commands for you:"
```

File: dev-env [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+echo "- htmlcov : Opens the test coverage results in your browser"
```
Comment (elemoine) [inline dev-env:70]:
```suggestion
echo "- htmlcov: Opens the test coverage results in your browser"
```

File: dev-env [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+echo "- htmldoc : Opens the locally built sphinx documentation in your browser"
```
Comment (elemoine) [inline dev-env:71]:
```suggestion
echo "- htmldoc: Opens the locally built sphinx documentation in your browser"
```

File: dev-env [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+echo "- check-lint: shorthand for tox -e check-lint (validates code style)"
+echo "- format: shorthand for tox -e format (rewrite files according to code style)"
```
Comment (elemoine) [inline dev-env:73]:
```suggestion
echo "- format: shorthand for tox -e format (rewrites files according to code style)"
```

File: dev-env (hunk 2 above)
Comment (ewjoachim) [reply dev-env:17]: TIL !

**Pull request merged**